### ISSUE TITLE
Rest High Level client putMapping operation doesn't respect IndicesOptions.lenientExpandOpen()

### ISSUE THREAD

**Issue by Evgueni-Gordienko** [elastic/elasticsearch#57045]
**Elasticsearch version** (`bin/elasticsearch --version`): the latest

**JVM version** (`java -version`): 1.8

**OS version** (`uname -a` if on a Unix-like system): Mas OS

**Description of the problem including expected versus actual behavior**:
When performing _putMapping_ operation via Rest High Level Client and setting indices option to _lenientExpandOpen_  an _ElasticsearchStatusException_ is thrown in case when no indices  were found. I expect no error to be returned.

**Steps to reproduce**:
Just perform something like this:
```
esRestClient.indices()
                        .putMapping(new PutMappingRequest(indices)
                                            .indicesOptions(IndicesOptions.lenientExpandOpen())
                                            .type(type)
                                            .source(mappings),
                                    RequestOptions.DEFAULT);
```

The root cause is in _org.elasticsearch.client.IndicesRequestConverters_, line 134.
For some reason, for the _putMapping_ method there is no line like the following:
`parameters.withIndicesOptions(putMappingRequest.indicesOptions());` which is present in many other methods of the class.

Because of that the resulting URL doesn't include the following parameters:
`?ignore_unavailable=true,allow_no_indices=true,expand_wildcards_open=true,expand_wildcards_closed=false`

And hence, I am getting back index_not_found_exception.

**Comment by cbuescher** [comment]
@Evgueni-Gordienko thanks for raising this, good catch and spot-on analysis. I opened a PR to fix this in the next 7.7 patch version.

**Comment by Evgueni-Gordienko** [comment]
@cbuescher sure, thanks for scheduling the fix.
